gapis/resolve/dependencygraph2/graph_visualization/graph_visualization.go

```diff
@@ -16,17 +16,208 @@ package graph_visualization
 
 import (
 	"context"
+	"fmt"
 	"github.com/google/gapid/core/math/interval"
 	"github.com/google/gapid/gapis/api"
 	"github.com/google/gapid/gapis/capture"
 	"github.com/google/gapid/gapis/resolve/dependencygraph2"
 )
 
-func GetGraphVisualizationFromCapture(ctx context.Context, p *capture.Capture) ([]byte, error) {
+const (
+	VK_BEGIN_COMMAND_BUFFER    = "vkBeginCommandBuffer"
+	VK_CMD_BEGIN_RENDER_PASS   = "vkCmdBeginRenderPass"
+	VK_CMD_NEXT_SUBPASS        = "vkCmdNextSubpass"
+	VK_COMMAND_BUFFER          = "vkCommandBuffer"
+	VK_RENDER_PASS             = "vkRenderPass"
+	VK_SUBPASS                 = "vkSubpass"
+	VK_END_COMMAND_BUFFER      = "vkEndCommandBuffer"
+	VK_CMD_END_RENDER_PASS     = "vkCmdEndRenderPass"
+	VK_CMD_DRAW_INDEXED        = "vkCmdDrawIndexed"
+	VK_CMD_DRAW                = "vkCmdDraw"
+	COMMAND_BUFFER             = "commandBuffer"
+	MAXIMUM_LEVEL_IN_HIERARCHY = 10
```

Comment:
These shouldn't be outside of the Vulkan package. 
In general we keep the knowledge of an API confined to the package where at all possible.

Reply:
Yes, I think it makes sense to abstract away this hierarchy logic into something like an interface providing `getCommandLabel`, which could be implemented by `vulkan.API`.

Reply:
Okay, I moved this into //gapis/api and //gapis/api/vulkan. I put the functions and structures such that the main code (//gapis/resolve/dependencygraph2/graph_visualization/..) depends only from //gapis/api considering future changes in the next commits.